src/App.tsx

```diff
@@ -1,24 +1,212 @@
+        return todos;
+    }
+  }, [todos, filter]);
+
+  useEffect(() => {
+    todoService.getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to download todos');
+      });
+  }, []);
+
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout | undefined;
+
+    if (errorMessage !== '') {
+      const clearErrorMessage = () => setErrorMessage('');
+
+      if (timeoutId) {
+        clearTimeout(timeoutId);
+      }
+
+      timeoutId = setTimeout(clearErrorMessage, 3000);
+    }
+
+    return () => {
+      if (timeoutId) {
+        clearTimeout(timeoutId);
+      }
+    };
+  }, [errorMessage]);
+
+  const allTodoComplited = useMemo(() => {
+    return todos.every(todo => todo.completed) && todos.length !== 0;
+  }, [todos]);
+
+  const uncomplitedTodos = useMemo(() => (
+    todos.filter(todo => !todo.completed).length
+  ), [todos]);
+
+  const complitedTodos = useMemo(() => (
+    todos.filter(todo => todo.completed).length
+  ), [todos]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {todos.length !== 0 && (
+            <button
+              type="button"
+              className={classNames(
+                'todoapp__toggle-all',
+                { active: allTodoComplited },
+              )}
+            />
+          )}
+
+          <NewTodo
+            addTodo={addTodo}
+            query={query}
+            setQuery={setQuery}
+            isAdding={isAdding}
+          />
+        </header>
+
+        <TodoList
+          visibleTodos={visibleTodos}
+          deleteTodo={deleteTodo}
+          loading={loading}
+          tempTodo={tempTodo}
+        />
+
+        {todos.length !== 0 && (
+          <footer className="todoapp__footer">
+            <span className="todo-count">
+              {`${uncomplitedTodos} items left`}
+            </span>
+
+            <Filter
+              filter={filter}
+              setFilter={setFilter}
+            />
+
+            <button
+              type="button"
+              className="todoapp__clear-completed"
+              disabled={complitedTodos === 0}
+              onClick={clearCompletedTodos}
+            >
+              Clear completed
+            </button>
+          </footer>
+        )}
+      </div>
+
+      <div className={classNames(
+        'notification is-danger is-light has-text-weight-normal',
+        { hidden: errorMessage.length === 0 },
```

Comment:
```suggestion
        { hidden: !errorMessage },
```
empty string is a falsy value: https://www.freecodecamp.org/news/falsy-values-in-javascript/